scripts/pre_commit_linter.py

```diff
@@ -1927,6 +1934,146 @@ def _check_for_copyright_notice(self):
+                        if line_in_concern.endswith('/'):
+                            line_in_concern = line_in_concern[:-1]
+                        # The following condition checks whether the specified
+                        # path exists in the codebase or not. The CODEOWNERS
+                        # syntax has paths starting with '/' which refers to
+                        # full path relative to root, but python glob module
+                        # does not conform to this logic and literally matches
+                        # the '/' character. Therefore the leading '/' has to
+                        # be removed for glob patterns to match correctly.
+                        if not glob.glob(line_in_concern.replace('/', '', 1)):
+                            print ('%s --> Pattern on line %s doesn\'t match '
+                                   'any file or directory' % (
+                                       codeowner_file, line_num))
+                            failed = True
+                        # Checks if the path is the full path relative to the
+                        # root oppia directory. Patterns starting with '/' are
+                        # considered relative to root whereas patterns starting
+                        # with './' are relative to the .github directory.
+                        if (not line_in_concern.startswith('/') and
+                                not './' +
+                                line_in_concern in CODEOWNER_FILE_PATHS):
+                            print ('%s --> Pattern on line %s is invalid. Use '
+                                   'full path relative to the root directory'
+                                   % (codeowner_file, line_num))
+                            failed = True
+                        # The double asterisks pattern is supported by the
+                        # CODEOWNERS syntax but not the glob in Python 2.
+                        # The following condition checks this.
+                        if '**' in line_in_concern:
+                            print ('%s --> Pattern on line %s is invalid. '
+                                   '\'**\' wildcard not allowed' % (
+                                       codeowner_file, line_num))
+                            failed = True
+                        # The following list is being populated with the
+                        # paths in the CODEOWNERS file with the removal of the
+                        # leading '/' to aid in the glob pattern matching in
+                        # the next part of the check wherein the valid patterns
+                        # are used to check if they cover the entire codebase.
+                        path_patterns.append(line_in_concern.replace(
+                            '/', '', 1))
+
+            # Checks that every dir/file is covered under CODEOWNERS.
+            for root, _, filename in os.walk('.'):
+                for file_name in filename:
```

Comment:
This reads terribly. Shouldn't the second arg be plural? Why does one have an underscore and the other not?

Reply:
Changed. Thanks!